### PR TITLE
feat(gti): Add tool to retrieve collection rules

### PR DESCRIPTION
**Pull request by XaviSamos**
This pull request introduces a new tool to retrieve detection rules from a specific collection, enhancing the capabilities of the GTI integration.

### Summary

The new tool, `get_collection_rules`, allows users to fetch community and curated hunting rules associated with a collection ID. It supports filtering by rule type and limiting the number of community rules returned. This functionality provides valuable threat intelligence context by making relevant detection rules easily accessible.

The changes include:
- A new server tool `get_collection_rules` in `server/gti/gti_mcp/tools/collections.py`.
- Comprehensive unit tests in `server/gti/tests/test_tools.py` to ensure correctness and handle various edge cases.

### Diagram

The following diagram illustrates the workflow of the `get_collection_rules` tool:

```mermaid
graph TD
    subgraph "get_collection_rules Workflow"
        A[Start] --> B{rule_types provided?};
        B -- No --> C[Default to all rule types];
        B -- Yes --> D[Use provided rule_types];
        C --> E;
        D --> E;
        E -- Community rules requested? --> F[Get collection aggregations];
        F --> G[Get top N community rules];
        G --> H[Fetch rule details];
        H --> I[Add to results];
        E -- Curated rules requested? --> J[Get related hunting rulesets];
        J --> K[Fetch ruleset content];
        K --> L[Add to results];
        I --> M[End];
        L --> M;
    end
```

### CODE REVIEW

File: server/gti/gti_mcp/tools/collections.py [hunk 1]
```diff
@@ -637,3 +637,141 @@ async def get_collections_commonalities(collection_id: str, ctx: Context) -> str
...
+  async with vt_client(ctx) as client:
+    # Fetch community rules from aggregations if requested
+    if any(rt in rule_types for rt in ["crowdsourced_ids", "crowdsourced_sigma", "crowdsourced_yara"]):
+      try:
+        data = await client.get_async(f"/collections/{collection_id}?attributes=aggregations")
+        data = await data.json_async()
+        attributes = data.get("data", {}).get("attributes", {})
+        if attributes:
+          files_aggregations = attributes.get("aggregations", {}).get("files", {})
+          if files_aggregations:
```
Comment (dprolongo) [inline server/gti/gti_mcp/tools/collections.py]: same comment
